### PR TITLE
MMAS-Added: RoundedButton UI Component

### PR DESCRIPTION
**Pull request by AlbertoMApps**
- Added the component for generic buttons, rounded,  border stroke, inner icons, text and fonts.

image

### CODE REVIEW

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        Icon(
+            imageVector = iconLeft, contentDescription = "Icon left"
+        )
+        Text(
+            modifier = Modifier.padding(start = padding),
+            text = title,
+            textAlign = TextAlign.Left,
+            fontWeight = fontWeight,
+            fontStyle = fontStyle,
+            fontSize = fontSize
```
Comment (jchodev) [inline core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt]: We can implement the font size like 
style = MaterialTheme.typography.titleMedium <-- we should following the Material guideline

MaterialTheme.typography was included fontStyle  & fontSize  & fontWeight

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    title: String,
+    fontWeight: FontWeight,
+    fontStyle: FontStyle,
+    fontSize: TextUnit,
+    padding: Dp,
+    onClick: () -> Unit = {}
+) {
+
+    Button(
+        onClick = onClick, shape = shape, border = border, colors = buttonColors
```
Comment (jchodev) [inline core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt:39]: we can implement the shape with https://m3.material.io/styles/shape/overview

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    ) {
+        Icon(
+            imageVector = iconLeft, contentDescription = "Icon left"
```
Comment (jchodev) [inline core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt]: content description pass from outside

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt (hunk 1 above)
Comment (AlbertoMApps) [reply core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt]: Yes, it should be the same.
I can remove the text size, fontWeight, etc for the text style instead.

File: core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt (hunk 2 above)
Comment (jchodev) [reply core/designsystem/src/main/java/com/jerryalberto/mmas/core/designsystem/component/RoundedButton.kt:39]: more information
image

**Review by jchodev — APPROVED** [review]
nice

**Pull request merged**